### PR TITLE
Move documentation inline for error types declared in base/base.jl.

### PR DESCRIPTION
**Pull request by xorJane**
I've moved the documentation inline for types `SystemError`, `ParseError`, `ArgumentError`, `KeyError`, `MethodError`,  `EOFError`, `DimensionMismatch`, `AssertionError`, `LoadError`, and `InitError`. Hope that helps!

### CODE REVIEW

File: base/base.jl [hunk 1]
```diff
@@ -4,15 +4,32 @@ type SystemError <: Exception
     prefix::AbstractString
     errnum::Int32
     extrainfo
+    @doc """
+       SystemError(prefix::AbstractString, [errno::Int32])
```
Comment (MichaelHatherly) [inline base/base.jl]: 3-space indent here, should be 4 to be recognised as a code block.

File: base/base.jl [hunk 2]
```diff
@@ -4,15 +4,32 @@ type SystemError <: Exception
...
+
+    A system call failed with an error code (in the `errno` global variable).
+    """ ->
```
Comment (MichaelHatherly) [inline base/base.jl]: Since this docstring doesn't apply specifically to the 3-arg constructor it would probably be better to just put it above the `type` itself. Same with the `DimensionMismatch` and `AssertionError` ones below as well.

**Comment by MichaelHatherly** [comment]
> Hope that helps!

:100:

**Comment by KristofferC** [comment]
`Base.jl` is getting thinner and thinner :)

**Comment by MichaelHatherly** [comment]
Only 7869 lines to go!

**Comment by xorJane** [comment]
Great, I made the modifications suggested. Thank you for the feedback @MichaelHatherly!

**Comment by kshyatt** [comment]
This was on my to-do list and you beat me! Nice work! 💯

File: base/base.jl [hunk 3]
```diff
@@ -21,39 +38,79 @@ end
...
+
+The asserted condition did not evaluate to `true`.
+Optional argument `msg` is a descriptive error string.
+"""
 type AssertionError <: Exception
     msg::AbstractString
-
-    AssertionError() = new("")
     AssertionError(msg) = new(msg)
+    AssertionError() = new("")
```
Comment (MichaelHatherly) [inline base/base.jl]: Changing the definition order probably won't affect anything, but best to avoid unrelated changes if possible. Other than that the PR looks go to me, thanks.

**Comment by kshyatt** [comment]
Also, @xorJane, this is your second Julia PR, right? (Belated) Welcome to Julia if so!

File: base/base.jl (hunk 3 above)
Comment (xorJane) [inline base/base.jl]: Reordered!

**Comment by xorJane** [comment]
Thanks @kshyatt !
